src/ppom-controller.ts

```diff
@@ -80,29 +71,43 @@ type PPOMVersionResponse = PPOMFileVersion[];
  * @type PPOMControllerState
  *
  * Controller state
- * @property lastChainId - ChainId for which files were last updated.
- * @property newChainId - ChainIf of currently selected network.
+ * @property chainId - ID of current chain.
+ * @property chainIdCache - Array of chainId and time it was last visited.
  * @property versionInfo - Version information fetched from CDN.
  * @property storageMetadata - Metadata of files storaged in storage.
  * @property providerRequestLimit - Number of requests in last 5 minutes that PPOM can make.
  * @property providerRequests - Array of timestamps in last 5 minutes when request was made from PPOM to provider.
  */
 export type PPOMControllerState = {
-  lastChainId: string;
-  newChainId: string;
+  // chainId of currently selected network
+  chainId: string;
```

Comment:
I appreciate we can extract this from the network change callback, but the standard elsewhere in many controllers is to use a callback to obtain this value whenever needed. Would that add consistency and simplicity here as we could minimise the state a little?

Reply:
We anyways need network change callback to create `chainIdCache`. Having chainId in state also enable this condition:

```
if (id === this.state.chainId) {
        return;
      }
```